### PR TITLE
MM-23898: Make deployments idempotent

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

- For Grafana datasource creation, we check for 409 conflict code.
  We also make the POST request with a timeout
- For system admin creation, we check if the user already exists.
- For ltagent init, we check if any resource already exists.
- We also disable auto_minor_version upgrade because this can clash
if the user runs deployment again, and by that time, AWS would have
upgraded the DB.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23898

### CODE REVIEW

**Comment by agnivade** [comment]
> If a deployment succeeds and the user tries a second time to deploy without previously having destroyed, we should return fast with an error or a message saying it's all good.

How do you propose we do that ? We don't persist any state at an application level about the deployment except terraform managing it's own internal state.

File: deployment/terraform/create.go [hunk 1]
```diff
@@ -294,6 +294,9 @@ func (t *Terraform) createAdminUser(extAgent *ssh.ExtAgent, output *terraformOut
 		return err
 	}
 	if out, err := sshc.RunCommand(cmd); err != nil {
+		if strings.Contains(string(out), "account with that username already exists") {
+			return nil
+		}
```
[the remark this answers is not in the document]
Comment (agnivade) [reply deployment/terraform/create.go:299]: I had thought about it. The problem is that deleting a user requires extra confirmation steps. And to do that, we need to do an "echo YES | delete_command". It's still hacky and unsafe. Hence I went with this.

Or did you want to use the client4 library to delete the user first ? At which point, we might as well create the user using the library too.

This seemed like a reasonable compromise of sorts.

**Comment by streamer45** [comment]
> How do you propose we do that ? We don't persist any state at an application level about the deployment except terraform managing it's own internal state.

That's where I wanted to get at. Can we figure out from a Terraform state whether the deployment succeeded?

File: deployment/terraform/create.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (streamer45) [reply deployment/terraform/create.go:299]: I've actually misread the code. We go ahead with the deployment if the sysadmin exists so we are good. No need to change this.

**Pull request merged**